highlevel/robot/gateway/motor.py

```diff
@@ -1,73 +1,156 @@
 """
 Motor gateway module.
 """
-from dataclasses import dataclass
 
 from highlevel.adapter.socket import SocketAdapter
 from highlevel.logger import LOGGER
-from highlevel.robot.entity.type import TickPerSec
+from highlevel.robot.entity.configuration import Configuration
+from highlevel.robot.entity.type import TickPerSec, Tick
+from highlevel.util.filter.pid import PIDConstants
 from proto.gen.python.outech_pb2 import BusMessage, MoveWheelAtSpeedMsg, PIDCoefficients, \
-    PIDConfigMsg
-
-
-@dataclass(frozen=True)
-class PIDValues:
-    """
-    Structure holding a PID's constants
-    """
-    k_p: float
-    k_i: float
-    k_d: float
-
-
-FIXED_POINT_COEF = 2**16
+    PIDConfigMsg, WheelPositionTargetMsg, WheelControlModeMsg, WheelTolerancesMsg, WheelPWMMsg
 
 
 class MotorGateway:
     """
     Motor gateway.
     """
-    def __init__(self, motor_board_adapter: SocketAdapter):
+    def __init__(self, motor_board_adapter: SocketAdapter,
+                 configuration: Configuration):
+        self.configuration = configuration
         self.motor_board_adapter = motor_board_adapter
+        self.pid_speed_left = PIDConstants(0, 0, 0)
+        self.pid_speed_right = PIDConstants(0, 0, 0)
+        self.pid_position_left = PIDConstants(0, 0, 0)
+        self.pid_position_right = PIDConstants(0, 0, 0)
 
     async def _send_message(self, message: BusMessage) -> None:
+        """
+        Serializes and sends sends a protobuf BusMessage through an adapter.
+        """
         payload = message.SerializeToString()
         await self.motor_board_adapter.send(payload)
 
-    async def set_speed(self, left_speed: TickPerSec,
-                        right_speed: TickPerSec) -> None:
+    def _pid_constants_to_proto(
+            self, pid_constants: PIDConstants) -> PIDCoefficients:
+        """
+        Converts a PIDConstants structure to a Protobuf PIDCoefficients sub message.
+        Applies a scaling factor to each value.
+        """
+        scale_factor = self.configuration.pid_scale_factor
+        update_rate = self.configuration.motor_update_rate
+        max_value_on_motor_board = (2**32) - 1
+        if pid_constants.k_i * scale_factor / update_rate >= max_value_on_motor_board:
+            LOGGER.get().error('pid_to_proto_ki_too_large',
```

Comment:
raise an exception? It's probably dangerous to silently fail here

Reply:
It is supposed to log an error, that's not silent.

Reply:
Ok but why not interrupt the flow of execution?

Is that an error that is expected to happen in real conditions?